### PR TITLE
Added gripper, controller and execution node

### PR DESCRIPTION
**Pull request by alex07zzz**
- Modified the gripper model
- Merged trajectory planner with robot execution
- Separated robot and robot rail planning
- Resolved tool planner node
- The "goal_abortion" code shows a common abortion when executing the trajectory. This abortion often happens when the object is attached to the gripper. The "robot_execution.py" ignored this features.

### CODE REVIEW

File: gilbreth_gazebo/config/robot_controller.yaml [hunk 1]
```diff
@@ -9,13 +9,20 @@ robot_controller:
...
+      robot_wrist_1_joint: {trajectory: 0.1, goal: 0.1}
+      robot_wrist_2_joint: {trajectory: 0.1, goal: 0.1}
+      robot_wrist_3_joint: {trajectory: 0.1, goal: 0.1}
+  gains:
+      robot_shoulder_pan_joint: {p: 10000, d: 150, i: 0, i_clamp: 1}
+      robot_shoulder_lift_joint: {p: 50000, d: 150, i: 10, i_clamp: 50}
+      robot_elbow_joint: {p: 50000, d: 150, i: 1, i_clamp: 25}
+      robot_wrist_1_joint: {p: 100, d: 5, i: 0, i_clamp: 1}
+      robot_wrist_2_joint: {p: 75, d: 2, i: 0, i_clamp: 1}
+      robot_wrist_3_joint: {p: 25, d: 1, i: 0, i_clamp: 1}
```
Comment (jrgnicho) [inline gilbreth_gazebo/config/robot_controller.yaml:25]: Out of curiosity, how did you arrive at these gain values?

File: gilbreth_gazebo/launch/gilbreth_environment.launch [hunk 2]
```diff
@@ -39,9 +39,11 @@
 
   <rosparam command="load" file="$(find gilbreth_gazebo)/config/joint_state_controller.yaml"/>
   <rosparam command="load" file="$(find gilbreth_gazebo)/config/robot_rail_controller.yaml"/>
+  <rosparam command="load" file="$(find gilbreth_gazebo)/config/robot_controller.yaml"/>
   <node name="gilbreth_controller_spawner" pkg="controller_manager" type="spawner"
     args="joint_state_controller robot_rail_controller"/>
-
+  <node name="gilbreth_controller_spawner2" pkg="controller_manager" type="spawner"
```
Comment (jrgnicho) [inline gilbreth_gazebo/launch/gilbreth_environment.launch:45]: Why is a second controller necessary?

File: gilbreth_grasp_planning/scripts/robot_execution_goal_abortion.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+        rb_exec.execute_robot()
+        rate.sleep()
+
+
+if __name__=='__main__':
+    try:
+        main(sys.argv)
+    except rospy.ROSInterruptException:
+        pass
+
```
Comment (jrgnicho) [inline gilbreth_grasp_planning/scripts/robot_execution_goal_abortion.py:283]: How is this script being used? The launch file above comments out the line where this gets called.

File: gilbreth_grasp_planning/scripts/test_pose_execution.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
```
Comment (jrgnicho) [inline gilbreth_grasp_planning/scripts/test_pose_execution.py:1]: I thought this whole script was already a part of the repo, it's showing as a new script.

**Review by jrgnicho — CHANGES_REQUESTED** [review]
Could you do a squash to remove the two "Merge ..." commits.  Let me know if you run into trouble doing that.

File: gilbreth_gazebo/config/robot_controller.yaml (hunk 1 above)
Comment (alex07zzz) [reply gilbreth_gazebo/config/robot_controller.yaml:25]: I quote the gains from ARIAC( https://bitbucket.org/osrf/ariac/src/4949e714932794b40513f6660d1af119148e2d67/osrf_gear/launch/ur10/arm_controller_ur10_custom.yaml?at=master&fileviewer=file-view-default).

File: gilbreth_gazebo/launch/gilbreth_environment.launch (hunk 2 above)
Comment (alex07zzz) [reply gilbreth_gazebo/launch/gilbreth_environment.launch:45]: This controller controls the robot_group joints, which does not contain linear actuator joint. I assume that each joint group has its own controller. I don't know if we can use one controller to control two group?

File: gilbreth_grasp_planning/scripts/robot_execution_goal_abortion.py (hunk 3 above)
Comment (alex07zzz) [reply gilbreth_grasp_planning/scripts/robot_execution_goal_abortion.py:283]: If you use this script to execute the robot. You can observe a weird behavior: 
After the robot successfully picks up the object, it starts to move to pick retreat pose. This execution process fails very often due to a GOAL_TOLERANCE_VIOLATION warning. This violation leads to an abortion of execution. 
I changed the controller yaml file to resolve the problem. Increasing the goal tolerance parameter will somehow decreases the chance of goal tolerance violation. But the abortion still happens very often.

File: gilbreth_grasp_planning/scripts/test_pose_execution.py (hunk 4 above)
Comment (alex07zzz) [reply gilbreth_grasp_planning/scripts/test_pose_execution.py:1]: This is a slightly different file I used to test the goal tolerance. I will delete it.

File: gilbreth_grasp_planning/scripts/robot_execution_goal_abortion.py (hunk 3 above)
Comment (alex07zzz) [reply gilbreth_grasp_planning/scripts/robot_execution_goal_abortion.py:283]: I don't know if it has something to do with the ros controller #317 issue. https://github.com/ros-controls/ros_controllers/issues/317

**Comment by alex07zzz** [comment]
Changes done.

**Comment by jrgnicho** [comment]
@alex07zzz  is this still relevant?

**Comment by jrgnicho** [comment]
@alex07zzz it looks like most of these changes have been captured in other PRs but before closing I want to make sure that is the case.  Is there anything in this PR you'd still like to merge?

**Pull request closed without merging**